services/genesis/templates.service.ts

```diff
@@ -1,14 +1,22 @@
 import { GlobalCore } from "@/core/module/module.types";
 import { createClient, SupabaseClient } from "@supabase/supabase-js";
 import config from "@/resources/utils/config";
+import MessageHandler from "@/core/message-handler";
+
+const messageHandler = MessageHandler.get();
+
+export interface TemplateField {
+  type: string;
+  description: string;
+}
 
 export interface Template {
   id: number;
   ownerId: string;
   name: string;
   createdAt: string;
   preview: string;
-  fields: object; // Might want to define a more specific type for fields
+  fields: { [key: string]: TemplateField }; // Might want to define a more specific type for fields
 }
 
 class TemplateService {
```

Comment:
This class is call `SupabaseTempalteService` create the generic Interface `TemplateService`